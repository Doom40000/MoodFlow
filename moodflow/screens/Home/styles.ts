import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'space-around',
    alignItems: 'center',
    paddingTop: 20,
    backgroundColor: '#69ABD1',
  },
  quoteContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: -200,
  },
  // quoteHeader: {
  //   paddingBottom: 10,
  // },
  quoteHeaderText: {
    color: 'white',
    backgroundColor: '#69ABD1',
    fontSize: 30,
    alignSelf: 'flex-start',
    paddingLeft: 20,
    fontWeight: 'bold',
  },
  quoteText: {
    color: 'white',
    fontSize: 28,
    marginHorizontal: 25,
    fontStyle: 'italic',
    backgroundColor: '#348EC2',
    padding: 20,
    paddingVertical: 40,
    borderRadius: 20,
  },
  iconContainer: {
    flex: 1,
    width: '75%',
    marginBottom: 15,
    justifyContent: 'center',
    alignItems: 'center',
  },
  iconText: {
    color: 'white',
    textAlign: 'center',
  },
  quote1: {
    alignSelf: 'flex-start',
  },
  quote2: {
    alignSelf: 'flex-end',
  },
});

export default styles;
