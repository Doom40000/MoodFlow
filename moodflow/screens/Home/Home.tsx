import React from 'react';
import { Text, View, Dimensions } from 'react-native';
import Svg, { Image } from 'react-native-svg';

import styles from './styles';

const imageSrc = require('../../assets/MoodFlowLogo.png');

const Home = () => {
  const { height, width } = Dimensions.get('window');

  return (
    <View style={styles.container}>
      <View style={{ flex: 1 }}>
        <Svg height={height} width={width}>
          <Image
            testID="Logo"
            href={imageSrc}
            width={width}
            height={height}
            preserveAspectRatio="xMidYMin meet"
            y={height / 7.5}
          />
        </Svg>
      </View>
      <View style={styles.quoteContainer}>
        <View style={styles.quoteHeader}>
          <Text style={styles.quoteHeaderText}>
            Here is your quote of the day:
          </Text>
        </View>
        {/* <View style={styles.quote}> */}
        <Text style={styles.quoteText}>
          “It's not about what it is, it's about what it can become.” Dr. Seuss
        </Text>
        {/* </View> */}
      </View>
    </View>
  );
};

export default Home;
