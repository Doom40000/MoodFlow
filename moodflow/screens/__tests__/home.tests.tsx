import { render } from '@testing-library/react-native';
import React from 'react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

import Home from '../Home/Home';

describe(Home, () => {
  it('should have multiple child components', () => {
    const tree = renderer.create(<Home />).toJSON() as ReactTestRendererJSON;
    expect(tree.children?.length).toBeGreaterThan(0);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the quote header text', () => {
    const { getByText } = render(<Home />);
    const textElement = getByText('Here is your quote of the day:');
    expect(textElement).toBeDefined();
  });

  it('should render the App logo', () => {
    const { getByTestId } = render(<Home />);
    const logoImage = getByTestId('Logo');
    expect(logoImage).toBeDefined();
  });
});
