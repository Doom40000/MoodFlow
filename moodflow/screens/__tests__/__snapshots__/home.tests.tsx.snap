// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "paddingTop": 20,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSVGSvgView
      bbHeight="1334"
      bbWidth="750"
      focusable={false}
      height={1334}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 1334,
            "width": 750,
          },
        ]
      }
      width={750}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGImage
          align="xMidYMin"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          height="1334"
          meetOrSlice={0}
          src={null}
          testID="Logo"
          width="750"
          x="0"
          y="177.86666666666667"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 2,
      }
    }
  >
    <View
      style={
        {
          "paddingBottom": 10,
        }
      }
    >
      <Text
        style={
          {
            "backgroundColor": "#1EAED7",
            "color": "white",
            "fontSize": 22,
          }
        }
      >
        Here is your quote of the day:
      </Text>
    </View>
    <Text
      style={
        {
          "color": "white",
          "fontSize": 28,
          "fontStyle": "italic",
          "marginHorizontal": 25,
        }
      }
    >
      “It's not about what it is, it's about what it can become.” Dr. Seuss
    </Text>
  </View>
</View>
`;
